js/binance.js

```diff
@@ -1903,6 +1984,29 @@ module.exports = class binance extends Exchange {
         return result;
     }
 
+    setMarkets (markets, currencies = undefined) {
+        super.setMarkets (markets, currencies);
+        const values = Object.values (this.markets);
+        let hasSpot = false;
+        for (let i = 0; i < values.length; i++) {
+            const entry = values[i];
+            hasSpot = hasSpot || entry['spot'];
+            if (hasSpot) {
+                break;
+            }
+        }
+        if (!hasSpot) {
+            // we add aliases for swap markets
+            for (let i = 0; i < values.length; i++) {
```

Comment:
Also, I find it weird that the number of markets !== 2 * number of symbols, we have 195 symbols and 387 markets, I will try to figure out why

Reply:
Just found out, two of them were futures, and one of them is a weird defi market that is not parsed correctly 
![image](https://user-images.githubusercontent.com/43336371/210062621-717121a9-65c5-4a14-a26f-1b53391d1d21.png)